tools/dotnet-linker/Steps/GenerateReferencesStep.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using Mono.Cecil;
+
+using Xamarin.Bundler;
+using Xamarin.Linker;
+
+namespace Xamarin {
+
+	public class GenerateReferencesStep : ConfigurationAwareStep {
+		private Symbols required_symbols = new Symbols ();
+		
+		protected override string Name { get; } = "Generate References";
+		protected override int ErrorCode { get; } = 2320;
+
+		protected override void TryEndProcess ()
+		{
+			base.TryEndProcess ();
+
+			var items = new List<MSBuildItem> ();
+			var file = Path.Combine (Configuration.CacheDirectory, $"references.mm");
+			if (Configuration.Target.GenerateReferencingSource (file, required_symbols) != null) {
+				var item = new MSBuildItem { Include = file };
+				items.Add (item);
+			}
+			Configuration.WriteOutputForMSBuild ("_ReferencesFile", items);
+		}
+
+		protected override void TryProcessAssembly (AssemblyDefinition assembly)
+		{
+			base.TryProcessAssembly (assembly);
+
+			if (!assembly.MainModule.HasTypes)
+				return;
+
+			var hasSymbols = false;
+			if (assembly.MainModule.HasModuleReferences) {
+				hasSymbols = true;
+			} else if (assembly.MainModule.HasTypeReference (Namespaces.Foundation + ".FieldAttribute")) {
```

Comment:
`"Foundation.FieldAttribute"` can be used here

The `Namespaces` code was required when we supported classic (`MonoTouch.Foundation` and `MonoMac.Foundation`)